File: workspace_js/services/js-executor/src/bootstrap/health.ts
import type { Application, ConfigModule } from '@mapexos/microservices';

import { container } from 'tsyringe';

import {
	RedisService as BaseRedisService,
	NatsBus,
	MinIOClient,
	REDIS_SERVICE_TOKEN,
	NATS_BUS_TOKEN,
} from '@mapexos/infrastructure';

import {
	registerHealthRoutes,
	RedisAdapter,
	NATSAdapter,
	MinIOAdapter,
} from '@mapexos/microservices';

// InitHealth registers the /health endpoint with all infrastructure checkers.
/** Registers health check endpoint with Redis, NATS, and MinIO checkers. */
export function initHealth(app: Application, configModule: ConfigModule) {
	const serviceName = configModule.get('service_name') as string;
	const serviceVersion = configModule.get('service_version') as string;

	const redisService = container.resolve<BaseRedisService>(REDIS_SERVICE_TOKEN);
	const natsBus = container.resolve<NatsBus>(NATS_BUS_TOKEN);
	const minioAssetsClient = container.resolve<MinIOClient>('MinIOAssetsClient');
	const minioTemplatesClient = container.resolve<MinIOClient>('MinIOTemplatesClient');
	const minioBytecodeClient = container.resolve<MinIOClient>('MinIOBytecodeClient');

	registerHealthRoutes(app, {
		serviceName,
		version: serviceVersion,
		cacheTTL: 10_000,
		timeout: 5_000,
	}, [
		{ checker: new RedisAdapter(redisService, 'app'), critical: true },
		{ checker: new NATSAdapter(natsBus, 'core'), critical: true },
		{ checker: new MinIOAdapter(minioAssetsClient, 'assets'), critical: false },
		{ checker: new MinIOAdapter(minioTemplatesClient, 'templates'), critical: false },
		{ checker: new MinIOAdapter(minioBytecodeClient, 'bytecode'), critical: false },
	]);
}

File: workspace_go/services/assets/src/bootstrap/health.go
package bootstrap

import (
	"time"

	"github.com/gofiber/fiber/v2"
	"go.uber.org/dig"

	config "github.com/Mapex-Solutions/mapexGoKit/microservices/config"
	container "github.com/Mapex-Solutions/mapexGoKit/microservices/container"

	minioModel "github.com/Mapex-Solutions/mapexGoKit/infrastructure/minio"
	mongoManager "github.com/Mapex-Solutions/mapexGoKit/infrastructure/mongodb/manager"
	natsModel "github.com/Mapex-Solutions/mapexGoKit/infrastructure/nats"
	redisModel "github.com/Mapex-Solutions/mapexGoKit/infrastructure/redis"

	"github.com/Mapex-Solutions/mapexGoKit/microservices/http/health"
	"github.com/Mapex-Solutions/mapexGoKit/microservices/http/health/adapters"
)

// InitHealth registers the /health endpoint with all infrastructure checkers.
// Single NATS connection (core). Nothing else is opened or health-checked
// from this MS.
func InitHealth(c *dig.Container, app *fiber.App) {
	c.Invoke(func(params struct {
		container.In
		MongoMgr    *mongoManager.MongoManager
		RedisApp    *redisModel.RedisClient
		RedisShared *redisModel.RedisClient `name:"shared"`
		NATSCore    *natsModel.Client       `name:"core"`
		MinIOAssets *minioModel.MinIOClient `name:"assets"`
	}) {
		serviceName, _ := config.GetStringValue("service_name")
		serviceVersion, _ := config.GetStringValue("service_version")

		health.RegisterRoutes(app, health.Config{
			ServiceName: serviceName,
			Version:     serviceVersion,
			CacheTTL:    10 * time.Second,
			Timeout:     5 * time.Second,
		},
			health.CheckerConfig{Checker: adapters.NewMongoAdapter(params.MongoMgr), Critical: true},
			health.CheckerConfig{Checker: adapters.NewRedisAdapter(params.RedisApp, "app"), Critical: true},
			health.CheckerConfig{Checker: adapters.NewRedisAdapter(params.RedisShared, "shared"), Critical: true},
			health.CheckerConfig{Checker: adapters.NewNATSAdapter(params.NATSCore, "core"), Critical: true},
			health.CheckerConfig{Checker: adapters.NewMinIOAdapter(params.MinIOAssets, "assets"), Critical: false},
		)
	})
}
